src/Core/Style.js

```diff
@@ -155,16 +145,163 @@ function defineStyleProperty(style, category, name, value, defaultValue) {
 }
 
 /**
- * A Style is an object that defines the visual appearance of {@link
+ * @typedef {Object} StyleOptions
```

Comment:
To lighten this PR, this documentation could be moved to another PR?

Reply:
The documentation is mostly the documentation of what is been changed in this PR, so i don't really see the use of doing a pre PR with minime change before added all the other changed in this PR...